### PR TITLE
doc: Improve contibutor doc formatting and readability

### PR DESCRIPTION
**Pull request by kirankashyap**
1. Converted types of refactoring into a numbered list
2. Removed readability

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -173,11 +173,15 @@ in the future, they may be removed by the Repository Maintainer.
...
-behaviour of code within the pull request (bugs must be preserved as is).
+There are three categories of refactoring namely,
+1. Code only moves
+2. Code style fixes
+3. Code refactoring
+
+In general, refactoring pull requests should not mix these
+three kinds of activity. Following this guideline will make refactoring pull requests easy to
+review and uncontroversial. In all cases, refactoring pull requests must not change the
+behaviour of code. _Even bugs must be preserved as is._
```
Comment (emilengler) [inline CONTRIBUTING.md]: A sentence starting with `Even` looks a bit weird to me. Maybe move it back into brackets like it was before?

**Comment by emilengler** [comment]
Hello kirankashyap,
welcome to Bitcoin Core :)

First please be sure to include a prefix to your pull request. Please change it to "doc: Improve contributor doc formatting and readability"

Also because of your requested review from fanquake. Please use the request review function **only** if you really know that THIS SPECIFIC person should review your PR but your PR could be reviewed by probably anyone as it is only some formatting and language improvements.

File: CONTRIBUTING.md (hunk 1 above)
Comment (kirankashyap) [reply CONTRIBUTING.md]: updated the changes as suggested. thanks for the suggestions.

**Comment by kirankashyap** [comment]
Thanks for the comments. I have updated the title. 

I will make sure not to add reviewers hereafter.

**Comment by emilengler** [comment]
Also be sure to squash your commits. There is a guide for it in the CONTRIBUTING.md file

**Comment by kirankashyap** [comment]
> Also be sure to squash your commits. There is a guide for it in the CONTRIBUTING.md file

@emilengler just resubmitted a single commit instead of two.  Thanks for the guidance.

**Comment by emilengler** [comment]
No problem.
But keep in mind that the `doc:` prefix also needs to be in the commit text.
To edit your commit message do:
```
git rebase -i HEAD~1
```
then edit the `pick` to `reword`
After this adjust the commit message and save it.
Finally:
```
git push origin patch-1 -f
```

Contact me on IRC if you need some sort of live chat support :D

**Comment by kirankashyap** [comment]
@emilengler I just pushed the commit with revised commit text as suggested. Thanks for the suggestion

**Review by fanquake — COMMENTED** [review]
I don't think the changes here are necessarily an improvement. The final sentence still needs fixing, `of code(Even bugs must be preserved as is).`, and the lines now break the 80 character word wrapping for no reason.

If you're interested in getting started contributing to the project I'd suggest looking at the [good first issue](https://github.com/bitcoin/bitcoin/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) list.

**Pull request closed without merging**